mobile/native/src/orderbook.rs

```diff
@@ -109,6 +110,15 @@ pub fn subscribe(
                                 tracing::info!("Received a rollover request from orderbook.");
                                 event::publish(&EventInternal::BackgroundNotification(BackgroundTask::Rollover(TaskStatus::Pending)));
 
+                                // Sending a failed event with a delay. This event will be ignored by the UI if the task has
+                                // been successful in the meantime.
+                                tokio::spawn(async {
+                                    tokio::time::sleep(ROLLOVER_PROTOCOL_TIMEOUT).await;
+                                    event::publish(&EventInternal::BackgroundNotification(
+                                        BackgroundTask::RecoverDlc(TaskStatus::Failed),
+                                    ));
+                                });
+
```

Comment:
🚨 what's the purpose of this one? 

we've just got a request from orderbook about doing rollover, and suddenly we're sending a *failure* about a different task? 

this seems incredibly hacky and counter-intuitive. Did you try to send information about `BackgroundTask::Rollover` perhaps? 

Also, seeing that the closing a dialog window development panned out from the backend side, I'm quite concerned with the direction of it - the backend should absolutely not be in the position to decide on the presentation layer for 
to be honest, I hoped for having that information sent when we *know* we failed etc. 😒
this does not seem better than just a timeout on the window in flutter, but it is an awful lot more complicated.

Reply:
> 🚨 what's the purpose of this one?

To show a failed (timed out) rollover in the frontend, if it hasn't received a success before the timeout.

> we've just got a request from orderbook about doing rollover, and suddenly we're sending a failure about a different task?

yes it would get ignored by the frontend if it has already gotten a success message! 

> Also, seeing that the closing a dialog window development panned out from the backend side, I'm quite concerned with the direction of it - the backend should absolutely not be in the position to decide on the presentation layer for

I do agree, that this timeout code should not be in the backend. I'll move it to the frontend. 👍 

> to be honest, I hoped for having that information sent when we know we failed etc. 😒
this does not seem better than just a timeout on the window in flutter, but it is an awful lot more complicated.

It's unfortunately not that easy to know when we've failed. Running into a timeout is just the easiest thing to do first. There is actually an open ticket to address this properly. This was not the intention of that change, but rather to add a simple safeguard that the user can always use the app eventually. (even if the protocol failed / or did not finish)